### PR TITLE
HMOS TextCounter Library

### PR DESCRIPTION
**Pull request by priyankabhadrappanavar**

### CODE REVIEW

File: LICENSE_APACHE [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "{}"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright 2017 Kshitij Jain
```
Comment (samuel-abhishek) [inline LICENSE_APACHE]: Please use the APACHE LICENSE file that was shared.

File: entry/src/main/java/com/github/premnirmal/textcounterdemo/ParanormalSlice.java [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+package com.github.premnirmal.textcounterdemo;
+
+import ohos.aafwk.ability.AbilitySlice;
+import ohos.aafwk.content.Intent;
+import com.github.premnirmal.textcounter.CounterView;
+import com.github.premnirmal.textcounter.Formatter;
+import java.text.NumberFormat;
+import java.util.Locale;
+
+/**
```
Comment (samuel-abhishek) [inline entry/src/main/java/com/github/premnirmal/textcounterdemo/ParanormalSlice.java:26]: This java doc can be removed as sample app content is developed differently for HarmonyOS.

File: README.md [hunk 3]
```diff
@@ -1,94 +1,83 @@
...
 
-![](img/counter.gif)
+## Feature
+A simple subclass of harmony's Text that counts up or down. Customize the counting using the attributes below.
 
-## Download
+See the demo project for more implementation details.
 
-### Gradle
+<img src="https://github.com/priyankabb153/TextCounter/blob/main/screenshots/TextCounterDemo.gif" width="256">
```
Comment (samuel-abhishek) [inline README.md]: Make sure this local url is pointing to the gif present in this project's "screenshots" folder

File: textcounter/src/main/java/com/github/premnirmal/textcounter/CounterView.java [hunk 4]
```diff
@@ -0,0 +1,339 @@
...
+                    counterType = CounterType.DECIMAL;
+                    break;
+                case 2:
+                    counterType = CounterType.BOTH;
+                    break;
+                default:
+                    HiLog.debug(HI_LOG_LABEL, "default switch case in init");
+            }
+        } finally {
+            //finally block
```
Comment (samuel-abhishek) [inline textcounter/src/main/java/com/github/premnirmal/textcounter/CounterView.java]: Can remove this finalyy block, as there is no implementation present in the block

File: LICENSE_APACHE (hunk 1 above)
Comment (priyankabhadrappanavar) [reply LICENSE_APACHE]: Done

File: entry/src/main/java/com/github/premnirmal/textcounterdemo/ParanormalSlice.java (hunk 2 above)
Comment (priyankabhadrappanavar) [reply entry/src/main/java/com/github/premnirmal/textcounterdemo/ParanormalSlice.java:26]: Done

File: README.md (hunk 3 above)
Comment (priyankabhadrappanavar) [reply README.md]: Done

File: textcounter/src/main/java/com/github/premnirmal/textcounter/CounterView.java (hunk 4 above)
Comment (priyankabhadrappanavar) [reply textcounter/src/main/java/com/github/premnirmal/textcounter/CounterView.java]: Done

File: README.md [hunk 5]
```diff
@@ -10,8 +10,7 @@ A simple subclass of harmony's Text that counts up or down. Customize the counti
 
 See the demo project for more implementation details.
 
-<img src="https://github.com/priyankabb153/TextCounter/blob/main/screenshots/TextCounterDemo.gif" width="256">
-
+<img src="screenshots/TextCounterDemo.gif" width="256">
```
Comment (samuel-abhishek) [inline README.md]: Can increase the text size in the gif. Currently it is too small for demo in tablet

**Review by samuel-abhishek — CHANGES_REQUESTED** [review]
Please check the review comment.

File: README.md (hunk 5 above)
Comment (priyankabhadrappanavar) [reply README.md]: Done

**Review by samuel-abhishek — APPROVED** [review]
LGTM

**Review by sharathks84 — APPROVED** [review]
OK to merge

**Review by Gleelakrishna — APPROVED** [review]
ok

**Pull request merged**